homeassistant/components/sensor/folder.py

```diff
@@ -58,21 +66,45 @@ class Folder(Entity):
 
     ICON = 'mdi:folder'
 
-    def __init__(self, folder_path, filter_term):
+    def __init__(self, folder_path, filter_term, hass):
         """Initialize the data object."""
         folder_path = os.path.join(folder_path, '')  # If no trailing / add it
         self._folder_path = folder_path   # Need to check its a valid path
         self._filter_term = filter_term
+        self._hass = hass
         self._number_of_files = None
+        self._files_dict = {}
         self._size = None
         self._name = os.path.split(os.path.split(folder_path)[0])[1]
         self._unit_of_measurement = 'MB'
 
     def update(self):
         """Update the sensor."""
-        files_list = get_files_list(self._folder_path, self._filter_term)
-        self._number_of_files = len(files_list)
-        self._size = get_size(files_list)
+        current_files = get_files_list(self._folder_path, self._filter_term)
+        self._number_of_files = len(current_files)
+        self._size = get_size(current_files)
+
+        previous_files = list(self._files_dict.keys())
+
+        for file_path in current_files:
+            file_name = os.path.split(file_path)[-1]
```

Comment:
What about subdirectories?

Reply:
Sorry, I think this comment was unclear. What I meant was that if you have directories in `current_files` then os.path.split would return an empty string for the tail section:

> if path ends in a slash, tail will be empty.

But apparently we're already just matching files, so that shouldn't be a problem (unless we're getting into case-insensitive file systems, but that's a whole other issue...)